### ISSUE TITLE
Re-negotiation between 1.1.1a and 1.0.2k version failed

### ISSUE THREAD

**Issue by zev-netizen** [openssl/openssl#10177]
I failed to renegotiate between version 1.1.1a and version 1.0.2k，as follows

OpenSSL 1.1.1a:
```
     openssl s_server -dtls1
```
OpenSSL 1.0.2k：
```
    openssl s_client -dtls1
```

Step 1：```Client input R```
Step 2：```Server input Enter```
then Client will get error messages：
```
SSL_connect:SSLv3 write client hello A
SSL3 alert write:fatal:illegal parameter
SSL_connect:error in SSLv3 read server hello A
140456473634448:error:140FD10F:SSL routines:DTLS1_GET_MESSAGE_FRAGMENT:bad length:d1_both.c:935
```

**Comment by jayaram-intel** [comment]
Hi @zhiweiw,

Just now I tried with OpenSSL 1.1.1a and 1.0.2k versions.
**Server side:**
/jmattax/OpenSSL_issues/10177/openssl-OpenSSL_1_1_1a/apps$ `./openssl s_server -dtls1`

> Using default temp DH parameters
> ACCEPT
> -----BEGIN SSL SESSION PARAMETERS-----
> MFYCAQECAwD+/wQCwBQEAAQwKZUg9ePtm9NCteCop/Wn2l+3cfAOycNv1zLKDu0c
> DMme1XXkdPPK53Q/Iak4RzdDoQYCBF2lXSyiBAICHCCkBgQEAQAAAA==
> -----END SSL SESSION PARAMETERS-----
> Shared ciphers:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES256-SHA:AES256-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:DHE-RSA-AES128-SHA:AES128-SHA
> Supported Elliptic Curve Point Formats: uncompressed:ansiX962_compressed_prime:ansiX962_compressed_char2
> Supported Elliptic Groups: P-256:P-521:brainpoolP512r1:brainpoolP384r1:P-384:brainpoolP256r1:secp256k1:B-571:K-571:K-409:B-409:K-283:B-283
> Shared Elliptic groups: P-256:P-521:P-384
> No server certificate CA names sent
> CIPHER is ECDHE-RSA-AES256-SHA
> Secure Renegotiation IS supported
>
>
> rRr
> Read BLOCK

**Client side:**
/jmattax/OpenSSL_issues/10177/openssl-OpenSSL_1_0_2k/apps$ `./openssl s_client -dtls1`

> CONNECTED(00000003)
> depth=0 C = UK, O = OpenSSL Group, OU = FOR TESTING PURPOSES ONLY, CN = Test Server Cert
> verify error:num=20:unable to get local issuer certificate
> verify return:1
> depth=0 C = UK, O = OpenSSL Group, OU = FOR TESTING PURPOSES ONLY, CN = Test Server Cert
> verify error:num=21:unable to verify the first certificate
> verify return:1
> Certificate chain
>  0 s:/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=Test Server Cert
>    i:/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=OpenSSL Test Intermediate CA
> Server certificate
> -----BEGIN CERTIFICATE-----
> MIID5zCCAs+gAwIBAgIJALnu1NlVpZ6zMA0GCSqGSIb3DQEBBQUAMHAxCzAJBgNV
> BAYTAlVLMRYwFAYDVQQKDA1PcGVuU1NMIEdyb3VwMSIwIAYDVQQLDBlGT1IgVEVT
> VElORyBQVVJQT1NFUyBPTkxZMSUwIwYDVQQDDBxPcGVuU1NMIFRlc3QgSW50ZXJt
> ZWRpYXRlIENBMB4XDTExMTIwODE0MDE0OFoXDTIxMTAxNjE0MDE0OFowZDELMAkG
> A1UEBhMCVUsxFjAUBgNVBAoMDU9wZW5TU0wgR3JvdXAxIjAgBgNVBAsMGUZPUiBU
> RVNUSU5HIFBVUlBPU0VTIE9OTFkxGTAXBgNVBAMMEFRlc3QgU2VydmVyIENlcnQw
> ggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDzhPOSNtyyRspmeuUpxfNJ
> KCLTuf7g3uQ4zu4iHOmRO5TQci+HhVlLZrHF9XqFXcIP0y4pWDbMSGuiorUmzmfi
> R7bfSdI/+qIQt8KXRH6HNG1t8ou0VSvWId5TS5Dq/er5ODUr9OaaDva7EquHIcMv
> vPQGuI+OEAcnleVCy9HVEIySrO4P3CNIicnGkwwiAud05yUAq/gPXBC1hTtmlPD7
> TVcGVSEiJdvzqqlgv02qedGrkki6GY4S7GjZxrrf7Foc2EP+51LJzwLQx3/JfrCU
> 41NEWAsu/Sl0tQabXESN+zJ1pDqoZ3uHMgpQjeGiE0olr+YcsSW/tJmiU9OiAr8R
> AgMBAAGjgY8wgYwwDAYDVR0TAQH/BAIwADAOBgNVHQ8BAf8EBAMCBeAwLAYJYIZI
> AYb4QgENBB8WHU9wZW5TU0wgR2VuZXJhdGVkIENlcnRpZmljYXRlMB0GA1UdDgQW
> BBSCvM8AABPR9zklmifnr9LvIBturDAfBgNVHSMEGDAWgBQ2w2yI55X+sL3szj49
> hqshgYfa2jANBgkqhkiG9w0BAQUFAAOCAQEAqb1NV0B0/pbpK9Z4/bNjzPQLTRLK
> WnSNm/Jh5v0GEUOE/Beg7GNjNrmeNmqxAlpqWz9qoeoFZax+QBpIZYjROU3TS3fp
> yLsrnlr0CDQ5R7kCCDGa8dkXxemmpZZLbUCpW2Uoy8sAA4JjN9OtsZY7dvUXFgJ7
> vVNTRnI01ghknbtD+2SxSQd3CWF6QhcRMAzZJ1z1cbbwGDDzfvGFPzJ+Sq+zEPds
> xoVLLSetCiBc+40ZcDS5dV98h9XD7JMTQfxzA7mNGv73JoZJA6nFgj+ADSlJsY/t
> JBv+z1iQRueoh9Qeee+ZbRifPouCB8FDx+AltvHTANdAq0t/K3o+pplMVA==
> -----END CERTIFICATE-----
> subject=/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=Test Server Cert
> issuer=/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=OpenSSL Test Intermediate CA
> No client certificate CA names sent
> Server Temp Key: ECDH, P-256, 256 bits
> SSL handshake has read 1998 bytes and written 610 bytes
> New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES256-SHA
> Server public key is 2048 bit
> Secure Renegotiation IS supported
> Compression: NONE
> Expansion: NONE
> No ALPN negotiated
> SSL-Session:
>     Protocol  : DTLSv1
>     Cipher    : ECDHE-RSA-AES256-SHA
>     Session-ID: 73FD4EC11FFC226DF6683EF802AAE39445B76284B22D5A4C7CE18A494E937A11
>     Session-ID-ctx: 
>     Master-Key: 299520F5E3ED9BD342B5E0A8A7F5A7DA5FB771F00EC9C36FD732CA0EED1C0CC99ED575E474F3CAE7743F21A938473743
>     Key-Arg   : None
>     PSK identity: None
>     PSK identity hint: None
>     SRP username: None
>     TLS session ticket lifetime hint: 7200 (seconds)
>     TLS session ticket:
>     0000 - 96 2e ad 72 2e 15 36 33-d5 01 69 c4 e1 83 6c 98   ...r..63..i...l.
>     0010 - 5f 48 dc ad c4 b6 4b 41-68 99 e8 fe 82 17 9f 93   _H....KAh.......
>     0020 - 6e 3e 41 57 ba 24 ed 3a-39 2e 35 1a 70 95 cc 79   n>AW.$.:9.5.p..y
>     0030 - 1b 4d f0 6a d6 a6 9c 82-7c be d3 bb 74 f1 1f 24   .M.j....|...t..$
>     0040 - 94 f9 5b 7c c2 27 36 4d-84 67 af 0c 5d 20 5f 9b   ..[|.'6M.g..] _.
>     0050 - 32 e9 d2 f7 ef c7 51 24-17 9e 96 d0 fc 73 1a a5   2.....Q$.....s..
>     0060 - 60 b1 28 fc fd 14 49 1e-37 b2 a9 72 8d ed 33 ba   `.(...I.7..r..3.
>     0070 - ff e0 ef d9 c1 3c fe 1c-67 79 ef 39 eb ed 97 02   .....<..gy.9....
>     0080 - 95 79 4b fd 96 84 2b 40-31 c6 fa f9 9e fc a6 36   .yK...+@1......6
>     0090 - 40 2a 89 23 05 00 95 d5-4b 8c a9 80 24 bc 74 d5   @*.#....K...$.t.
> 
>     Start Time: 1571118380
>     Timeout   : 7200 (sec)
>     Verify return code: 21 (unable to verify the first certificate)
> r
> R
> r
> R
> RENEGOTIATING
> depth=0 C = UK, O = OpenSSL Group, OU = FOR TESTING PURPOSES ONLY, CN = Test Server Cert
> verify error:num=20:unable to get local issuer certificate
> verify return:1
> depth=0 C = UK, O = OpenSSL Group, OU = FOR TESTING PURPOSES ONLY, CN = Test Server Cert
> verify error:num=21:unable to verify the first certificate
> verify return:1

I didn't seen Error messages mentioned in your previous comment.


Thanks
Jayaram Matta

**Comment by zhiweiw** [comment]
When the client and server establish a DTLS link, ***server side does not perform any operation***,

next，
> on the client side inputs R

then
> on the server sends data randomly.

client will report error

**Comment by jayaram-intel** [comment]
Hi @zhiweiw,

tried as mentioned above but **client not reporting any error**.

Note:
Tried it on OS: Ubuntu 16.04 (with Kernel: 5.2.14-050214-generic)

Thanks
Jayaram Matta

**Comment by zhiweiw** [comment]
hi @jayaram-intel , thank you for following up! My test env info as follows:

***OS: WSL - SUSE12SP3***
openssl config param: 
> openssl 1.1.1a: ./config no-shared --debug
openssl 1.0.2k: ./config no-shared -d

### 1. On Server side
```
dev:/mnt/d/Documents/code/openssl-OpenSSL_1_1_1a/apps # ./openssl s_server -dtls1
Using default temp DH parameters
ACCEPT
```
### 2. Client side
```
dev:/mnt/d/Documents/code/openssl-OpenSSL_1_0_2k/apps # ./openssl s_client -dtls1
WARNING: can't open config file: /usr/local/ssl/openssl.cnf
CONNECTED(00000003)
depth=0 C = UK, O = OpenSSL Group, OU = FOR TESTING PURPOSES ONLY, CN = Test Server Cert
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 C = UK, O = OpenSSL Group, OU = FOR TESTING PURPOSES ONLY, CN = Test Server Cert
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=Test Server Cert
   i:/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=OpenSSL Test Intermediate CA
---
Server certificate
-----BEGIN CERTIFICATE-----
MIID5zCCAs+gAwIBAgIJALnu1NlVpZ6zMA0GCSqGSIb3DQEBBQUAMHAxCzAJBgNV
BAYTAlVLMRYwFAYDVQQKDA1PcGVuU1NMIEdyb3VwMSIwIAYDVQQLDBlGT1IgVEVT
VElORyBQVVJQT1NFUyBPTkxZMSUwIwYDVQQDDBxPcGVuU1NMIFRlc3QgSW50ZXJt
ZWRpYXRlIENBMB4XDTExMTIwODE0MDE0OFoXDTIxMTAxNjE0MDE0OFowZDELMAkG
A1UEBhMCVUsxFjAUBgNVBAoMDU9wZW5TU0wgR3JvdXAxIjAgBgNVBAsMGUZPUiBU
RVNUSU5HIFBVUlBPU0VTIE9OTFkxGTAXBgNVBAMMEFRlc3QgU2VydmVyIENlcnQw
ggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDzhPOSNtyyRspmeuUpxfNJ
KCLTuf7g3uQ4zu4iHOmRO5TQci+HhVlLZrHF9XqFXcIP0y4pWDbMSGuiorUmzmfi
R7bfSdI/+qIQt8KXRH6HNG1t8ou0VSvWId5TS5Dq/er5ODUr9OaaDva7EquHIcMv
vPQGuI+OEAcnleVCy9HVEIySrO4P3CNIicnGkwwiAud05yUAq/gPXBC1hTtmlPD7
TVcGVSEiJdvzqqlgv02qedGrkki6GY4S7GjZxrrf7Foc2EP+51LJzwLQx3/JfrCU
41NEWAsu/Sl0tQabXESN+zJ1pDqoZ3uHMgpQjeGiE0olr+YcsSW/tJmiU9OiAr8R
AgMBAAGjgY8wgYwwDAYDVR0TAQH/BAIwADAOBgNVHQ8BAf8EBAMCBeAwLAYJYIZI
AYb4QgENBB8WHU9wZW5TU0wgR2VuZXJhdGVkIENlcnRpZmljYXRlMB0GA1UdDgQW
BBSCvM8AABPR9zklmifnr9LvIBturDAfBgNVHSMEGDAWgBQ2w2yI55X+sL3szj49
hqshgYfa2jANBgkqhkiG9w0BAQUFAAOCAQEAqb1NV0B0/pbpK9Z4/bNjzPQLTRLK
WnSNm/Jh5v0GEUOE/Beg7GNjNrmeNmqxAlpqWz9qoeoFZax+QBpIZYjROU3TS3fp
yLsrnlr0CDQ5R7kCCDGa8dkXxemmpZZLbUCpW2Uoy8sAA4JjN9OtsZY7dvUXFgJ7
vVNTRnI01ghknbtD+2SxSQd3CWF6QhcRMAzZJ1z1cbbwGDDzfvGFPzJ+Sq+zEPds
xoVLLSetCiBc+40ZcDS5dV98h9XD7JMTQfxzA7mNGv73JoZJA6nFgj+ADSlJsY/t
JBv+z1iQRueoh9Qeee+ZbRifPouCB8FDx+AltvHTANdAq0t/K3o+pplMVA==
-----END CERTIFICATE-----
subject=/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=Test Server Cert
issuer=/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=OpenSSL Test Intermediate CA
---
No client certificate CA names sent
Server Temp Key: ECDH, P-256, 256 bits
---
SSL handshake has read 1998 bytes and written 610 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES256-SHA
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : DTLSv1
    Cipher    : ECDHE-RSA-AES256-SHA
    Session-ID: B87E175CC0C4E5D977C4F7EC54DF35C2EB95FE85C3A53FAAA6AB51975234CE1B
    Session-ID-ctx:
    Master-Key: 4701743D4216DB1CC45CE6B541F530AECC2A6EC654ED632473C0C243F0B909404E1768ABBFFD56072CF8F04112BBE739
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - f5 17 8e 86 5e 68 45 bd-39 3a 6c 64 08 c2 39 38   ....^hE.9:ld..98
    0010 - 5b 8d 92 52 d1 8a 9c 6c-bc 63 ea 8c 3f c2 34 39   [..R...l.c..?.49
    0020 - 50 77 66 9a ee ad 95 3f-92 75 e1 a4 13 fd 68 3a   Pwf....?.u....h:
    0030 - e8 df 96 fa 9f e1 88 1f-02 8d 6a 27 d9 b6 86 ef   ..........j'....
    0040 - fe bb a4 39 db 67 1f 10-46 e5 91 17 32 e4 c9 02   ...9.g..F...2...
    0050 - db c0 6a e6 04 4a 50 79-46 3b b0 69 74 83 c4 9a   ..j..JPyF;.it...
    0060 - 62 ee 9c 19 8f fe 37 c5-c8 ef 9b 6f 22 05 51 51   b.....7....o".QQ
    0070 - d3 a8 de a8 d3 5d cd cf-d0 c9 53 7f fc 6c 7d 81   .....]....S..l}.
    0080 - 96 93 48 43 e8 fc b9 58-1f 63 32 3b 54 4f 67 a2   ..HC...X.c2;TOg.
    0090 - 94 e8 87 85 5a 12 0f 30-79 c1 9f 30 3b 8a 6a 9f   ....Z..0y..0;.j.

    Start Time: 1571181831
    Timeout   : 7200 (sec)
    Verify return code: 21 (unable to verify the first certificate)
---
```
### 3.  now, ***server side does not perform any operation***, and on the client side, click ***'R'***
```
dev:/mnt/d/Documents/code/openssl-OpenSSL_1_0_2k/apps # ./openssl s_client -dtls1
WARNING: can't open config file: /usr/local/ssl/openssl.cnf
CONNECTED(00000003)
depth=0 C = UK, O = OpenSSL Group, OU = FOR TESTING PURPOSES ONLY, CN = Test Server Cert
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 C = UK, O = OpenSSL Group, OU = FOR TESTING PURPOSES ONLY, CN = Test Server Cert
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=Test Server Cert
   i:/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=OpenSSL Test Intermediate CA
---
Server certificate
-----BEGIN CERTIFICATE-----
MIID5zCCAs+gAwIBAgIJALnu1NlVpZ6zMA0GCSqGSIb3DQEBBQUAMHAxCzAJBgNV
BAYTAlVLMRYwFAYDVQQKDA1PcGVuU1NMIEdyb3VwMSIwIAYDVQQLDBlGT1IgVEVT
VElORyBQVVJQT1NFUyBPTkxZMSUwIwYDVQQDDBxPcGVuU1NMIFRlc3QgSW50ZXJt
ZWRpYXRlIENBMB4XDTExMTIwODE0MDE0OFoXDTIxMTAxNjE0MDE0OFowZDELMAkG
A1UEBhMCVUsxFjAUBgNVBAoMDU9wZW5TU0wgR3JvdXAxIjAgBgNVBAsMGUZPUiBU
RVNUSU5HIFBVUlBPU0VTIE9OTFkxGTAXBgNVBAMMEFRlc3QgU2VydmVyIENlcnQw
ggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDzhPOSNtyyRspmeuUpxfNJ
KCLTuf7g3uQ4zu4iHOmRO5TQci+HhVlLZrHF9XqFXcIP0y4pWDbMSGuiorUmzmfi
R7bfSdI/+qIQt8KXRH6HNG1t8ou0VSvWId5TS5Dq/er5ODUr9OaaDva7EquHIcMv
vPQGuI+OEAcnleVCy9HVEIySrO4P3CNIicnGkwwiAud05yUAq/gPXBC1hTtmlPD7
TVcGVSEiJdvzqqlgv02qedGrkki6GY4S7GjZxrrf7Foc2EP+51LJzwLQx3/JfrCU
41NEWAsu/Sl0tQabXESN+zJ1pDqoZ3uHMgpQjeGiE0olr+YcsSW/tJmiU9OiAr8R
AgMBAAGjgY8wgYwwDAYDVR0TAQH/BAIwADAOBgNVHQ8BAf8EBAMCBeAwLAYJYIZI
AYb4QgENBB8WHU9wZW5TU0wgR2VuZXJhdGVkIENlcnRpZmljYXRlMB0GA1UdDgQW
BBSCvM8AABPR9zklmifnr9LvIBturDAfBgNVHSMEGDAWgBQ2w2yI55X+sL3szj49
hqshgYfa2jANBgkqhkiG9w0BAQUFAAOCAQEAqb1NV0B0/pbpK9Z4/bNjzPQLTRLK
WnSNm/Jh5v0GEUOE/Beg7GNjNrmeNmqxAlpqWz9qoeoFZax+QBpIZYjROU3TS3fp
yLsrnlr0CDQ5R7kCCDGa8dkXxemmpZZLbUCpW2Uoy8sAA4JjN9OtsZY7dvUXFgJ7
vVNTRnI01ghknbtD+2SxSQd3CWF6QhcRMAzZJ1z1cbbwGDDzfvGFPzJ+Sq+zEPds
xoVLLSetCiBc+40ZcDS5dV98h9XD7JMTQfxzA7mNGv73JoZJA6nFgj+ADSlJsY/t
JBv+z1iQRueoh9Qeee+ZbRifPouCB8FDx+AltvHTANdAq0t/K3o+pplMVA==
-----END CERTIFICATE-----
subject=/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=Test Server Cert
issuer=/C=UK/O=OpenSSL Group/OU=FOR TESTING PURPOSES ONLY/CN=OpenSSL Test Intermediate CA
---
No client certificate CA names sent
Server Temp Key: ECDH, P-256, 256 bits
---
SSL handshake has read 1998 bytes and written 610 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES256-SHA
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : DTLSv1
    Cipher    : ECDHE-RSA-AES256-SHA
    Session-ID: B87E175CC0C4E5D977C4F7EC54DF35C2EB95FE85C3A53FAAA6AB51975234CE1B
    Session-ID-ctx:
    Master-Key: 4701743D4216DB1CC45CE6B541F530AECC2A6EC654ED632473C0C243F0B909404E1768ABBFFD56072CF8F04112BBE739
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - f5 17 8e 86 5e 68 45 bd-39 3a 6c 64 08 c2 39 38   ....^hE.9:ld..98
    0010 - 5b 8d 92 52 d1 8a 9c 6c-bc 63 ea 8c 3f c2 34 39   [..R...l.c..?.49
    0020 - 50 77 66 9a ee ad 95 3f-92 75 e1 a4 13 fd 68 3a   Pwf....?.u....h:
    0030 - e8 df 96 fa 9f e1 88 1f-02 8d 6a 27 d9 b6 86 ef   ..........j'....
    0040 - fe bb a4 39 db 67 1f 10-46 e5 91 17 32 e4 c9 02   ...9.g..F...2...
    0050 - db c0 6a e6 04 4a 50 79-46 3b b0 69 74 83 c4 9a   ..j..JPyF;.it...
    0060 - 62 ee 9c 19 8f fe 37 c5-c8 ef 9b 6f 22 05 51 51   b.....7....o".QQ
    0070 - d3 a8 de a8 d3 5d cd cf-d0 c9 53 7f fc 6c 7d 81   .....]....S..l}.
    0080 - 96 93 48 43 e8 fc b9 58-1f 63 32 3b 54 4f 67 a2   ..HC...X.c2;TOg.
    0090 - 94 e8 87 85 5a 12 0f 30-79 c1 9f 30 3b 8a 6a 9f   ....Z..0y..0;.j.

    Start Time: 1571181831
    Timeout   : 7200 (sec)
    Verify return code: 21 (unable to verify the first certificate)
---
R

RENEGOTIATING
140669880370832:error:140FD10F:SSL routines:DTLS1_GET_MESSAGE_FRAGMENT:bad length:d1_both.c:935:
```
### 4.server receive alert message and close connection
```
dev:/mnt/d/Documents/code/openssl-OpenSSL_1_1_1a/apps # ./openssl s_server -dtls1
Using default temp DH parameters
ACCEPT
-----BEGIN SSL SESSION PARAMETERS-----
MFYCAQECAwD+/wQCwBQEAAQwRwF0PUIW2xzEXOa1QfUwrswqbsZU7WMkc8DCQ/C5
CUBOF2irv/1WByz48EESu+c5oQYCBF2mVQeiBAICHCCkBgQEAQAAAA==
-----END SSL SESSION PARAMETERS-----
Shared ciphers:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES256-SHA:AES256-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:DHE-RSA-AES128-SHA:AES128-SHA
Supported Elliptic Curve Point Formats: uncompressed:ansiX962_compressed_prime:ansiX962_compressed_char2
Supported Elliptic Groups: P-256:P-521:brainpoolP512r1:brainpoolP384r1:P-384:brainpoolP256r1:secp256k1:B-571:K-571:K-409:B-409:K-283:B-283
Shared Elliptic groups: P-256:P-521:P-384
---
No server certificate CA names sent
CIPHER is ECDHE-RSA-AES256-SHA
Secure Renegotiation IS supported
TIMEOUT occurred
TIMEOUT occurred
TIMEOUT occurred

ERROR
139964961326848:error:14102417:SSL routines:dtls1_read_bytes:sslv3 alert illegal parameter:ssl/record/rec_layer_d1.c:625:SSL alert number 47
shutting down SSL
CONNECTION CLOSED
```

### The screenshots are as follows
Client
![CLIEN](https://user-images.githubusercontent.com/3162314/66878004-8688aa80-efea-11e9-8f5b-1eaa7624cb8f.png)

Server:
![SERVER](https://user-images.githubusercontent.com/3162314/66878035-a7510000-efea-11e9-93b3-5a82a4ec490a.png)

**Comment by mattcaswell** [comment]
I just tried this out. I didn't quite get the same behaviour as you - but I *did* see those weird "TIMEOUT occurred" messages on the server side and a renegotiation did not properly take place. However after I built the server with the latest 1.1.1d code it all worked as expected. Therefore I suspect you are hitting a bug which has subsequently been fixed. Please upgrade to 1.1.1d and try again.

**Comment by Mengliuxx** [comment]
Do you know Where is this modification?

**Comment by mattcaswell** [comment]
Issue seems to have been fixed in the latest 1.1.1. Closing this.
